iOS/Flows/Conversation/Cells/ConversationMessageCellLayout.swift

```diff
@@ -0,0 +1,124 @@
+
+        // The higher the cell's index, the closer it is to the front of the message stack.
+        let stackIndex = totalItemsInSection - indexPath.item - 1
+
+        // Objects closer to the front of the stack should be brighter.
+        let backgroundBrightness = 1 - CGFloat(stackIndex) * 0.05
+        var backgroundColor: UIColor = indexPath.section == 0 ? .lightGray : .gray
+        backgroundColor = backgroundColor.color(withBrightness: backgroundBrightness)
+
+        attributes.backgroundColor = backgroundColor
+
+        // Only show text for the front most item in each section.
+        attributes.shouldShowText = stackIndex == 0
+
+        if indexPath.section == 0 {
+            // The first section should have a bubble tail on its first item
+            attributes.shouldShowTail = stackIndex == totalItemsInSection - 1
+            attributes.bubbleTailOrientation = .up
+        }
+
+        if indexPath.section == 1 {
+            // The second section should have a tail on its last item
+            attributes.shouldShowTail = stackIndex == 0
+            attributes.bubbleTailOrientation = .down
+        }
+    }
+
+    // MARK: - Custom Animations
+
+    private var insertingIndexPaths: [IndexPath] = []
+    private var deletingIndexPaths: [IndexPath] = []
+
+    override func prepare(forCollectionViewUpdates updateItems: [UICollectionViewUpdateItem]) {
+        super.prepare(forCollectionViewUpdates: updateItems)
+
+        self.insertingIndexPaths.removeAll()
+        self.deletingIndexPaths.removeAll()
+
+        for update in updateItems {
+            if let indexPath = update.indexPathBeforeUpdate, update.updateAction == .delete {
+                self.deletingIndexPaths.append(indexPath)
+            }
+
+            if let indexPath = update.indexPathAfterUpdate, update.updateAction == .insert {
+                self.insertingIndexPaths.append(indexPath)
+            }
+        }
+    }
+
+    override func initialLayoutAttributesForAppearingItem(at itemIndexPath: IndexPath)
+    -> UICollectionViewLayoutAttributes? {
+
+        let attributes = super.initialLayoutAttributesForAppearingItem(at: itemIndexPath)
+        // Because the user is dropping the message directly on the stack,
+        // the message should appear immediately, not fade in.
+        attributes?.alpha = 1
+        return attributes
+    }
+
+    override func finalLayoutAttributesForDisappearingItem(at itemIndexPath: IndexPath)
+    -> UICollectionViewLayoutAttributes? {
+
+        let attributes = super.finalLayoutAttributesForDisappearingItem(at: itemIndexPath)
+
+        if self.deletingIndexPaths.contains(itemIndexPath) {
+            // Shrink down the message we're deleting. This is also prevents the deleted cell from
+            // covering other views that are moving to take its place.
+            attributes?.transform = CGAffineTransform(scaleX: 0.01, y: 0.01)
+        }
```

Comment:
Items should appear immediately so it feels responsive. 

Items that are disappearing shrink down.